mlflow/models/evaluation/utils/trace.py

```diff
@@ -0,0 +1,117 @@
+import contextlib
+import inspect
+import logging
+from typing import Any
+
+import mlflow
+from mlflow.ml_package_versions import FLAVOR_TO_MODULE_NAME
+from mlflow.utils.autologging_utils import AUTOLOGGING_INTEGRATIONS, autologging_conf_lock
+
+_logger = logging.getLogger(__name__)
+
+
+def _kwargs_safe_invoke(func: callable, kwargs: dict[str, Any]):
```

Comment:
nit: (maybe just a matter of preference) We may want to put this method below so that private and public methods are ordered consistently.

Reply:
Agreed. I prefer the order but was just too lazy to format:p